### PR TITLE
Improvements for event data

### PR DESCRIPTION
**Pull request by denissdubinin**
- Add missing data for various events
- Add event for 404 page

### CODE REVIEW

File: src/scandipwa/app/component/GoogleTagManager/events/NotFound.event.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+     * Handler
+     */
+    handler() {
+        if (this.spamProtection(SPAM_PROTECTION_DELAY)) {
+            return;
+        }
+
+        this.pushEventData({
+            eventCategory: '404 pages',
+            eventAction: window.location.href,
```
Comment (denissdubinin) [inline src/scandipwa/app/component/GoogleTagManager/events/NotFound.event.js:43]: withRouter is not working in this case

File: src/scandipwa/app/component/GoogleTagManager/events/Purchase.event.js [hunk 2]
```diff
@@ -67,18 +67,18 @@ class Purchase extends BaseEvent {
...
             id: orderId,
-            tax: +roundPrice(tax_amount),
+            tax: +roundPrice(tax_amount || 0),
             coupon: coupon_code,
-            revenue: +roundPrice(grand_total),
-            shipping: +roundPrice(shipping_amount),
-            coupon_discount_amount: +roundPrice(discount_amount)
+            revenue: +roundPrice(grand_total || 0),
+            shipping: +roundPrice(shipping_amount || 0),
+            coupon_discount_amount: +roundPrice(discount_amount || 0)
```
Comment (yeegor) [inline src/scandipwa/app/component/GoogleTagManager/events/Purchase.event.js:81]: Consider setting the default values for these variables during the restructuring. Just as you did with the `coupon_code`. Or are the other things different?
E.g. 
```js
getActionFields(orderId = '', {
    tax_amount = 0, 
    grand_total = 0, 
    shipping_amount = 0, 
    discount_amount = 0, 
    coupon_code = ''
}) { ... }
```

File: src/scandipwa/app/component/GoogleTagManager/utils/Product.js [hunk 3]
```diff
@@ -65,11 +65,20 @@ export default class Product {
         return attribute_value;
     }
 
+    /**
+     * @param product
+     * @returns {*|number|bigint}
+     */
```
Comment (yeegor) [inline src/scandipwa/app/component/GoogleTagManager/utils/Product.js:71]: I doubt that this function returns a number.

File: src/scandipwa/app/component/GoogleTagManager/utils/Product.js [hunk 4]
```diff
@@ -106,21 +116,29 @@ export default class Product {
...
+            price_range: {
+                minimum_price: {
+                    final_price: {
                         value: discountValue = null
                     } = {}
                 } = {},
-                regularPrice: {
-                    amount: {
+                regular_price: {
+                    final_price: {
```
Comment (yeegor) [inline src/scandipwa/app/component/GoogleTagManager/utils/Product.js:141]: Are you 100% sure these changes are not project-specific? It seems they might be.

File: src/scandipwa/app/plugin/StoreModifications.plugin.js [hunk 5]
```diff
@@ -72,8 +84,13 @@ const afterRequestCustomerData = (args, callback, instance) => {
 
     return callback(...args)
         .then(result => {
-            transferGroupedProductsData(customer.id);
-            gtm.updateGroupedProductsStorageName(customer.id);
+            const CUSTOMER = 'customer';
```
Comment (yeegor) [inline src/scandipwa/app/plugin/StoreModifications.plugin.js:87]: Consider importing this from somewhere

File: src/scandipwa/app/plugin/query/ConfigQuery.plugin.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/base-theme
+ * @author Deniss Dubinins <denissd@scandiweb.com | info@scandiweb.com>
+ * @link https://github.com/scandipwa/base-theme
+ */
+
+import { PureComponent } from 'react';
+
+export class ConfigQueryPlugin extends PureComponent {
```
Comment (yeegor) [inline src/scandipwa/app/plugin/query/ConfigQuery.plugin.js:15]: You should not extend PureComponent with the plugin classes.
Also, FYI, I see that you actually don't need a plugin class here at all. We are shifting towards removing the classes from .plugin.js files when they don't serve some specific purpose, e.g. preserving the state.

**Review by yeegor — CHANGES_REQUESTED** [review]
Overall code quality is very good
Only minor changes are requested. Thank you!

**Pull request closed without merging**